components/net/http_loader.rs

```diff
@@ -1462,7 +1462,15 @@ fn http_network_fetch(
             _ => false,
         });
 
-    if !req_origin_in_timing_allow && !wildcard_present {
+    let is_same_origin = request.url_list.iter().any(|url| match request.origin {
```

Comment:
We should either call this `is_cross_origin` and check if any of the urls in the url list are not same-origin, or we should call it `is_same_origin` and check if _all_ of the urls in the url list are same-origin.

Reply:
Thanks for the review.

I've fixed it as following.

> check if all of the urls in the url list are same-origin.

and

```rust
let is_same_origin = request.url_list.iter().all(|url| match request.origin {
```